icu4c/source/common/uloc.cpp

```diff
@@ -1557,32 +1557,18 @@ uloc_openKeywords(const char* localeID,
         tmpLocaleID=localeID;
     }
 
-    /* Skip the language */
-    ulocimp_getLanguage(tmpLocaleID, &tmpLocaleID, *status);
+    ulocimp_getSubtags(
```

Comment:
Why does the old and new code here bother to parse/identify the subtags at all, when eventually it just looks for `@` and parses the keywords from there?